GNUmakefile

```diff
@@ -4,28 +4,20 @@ PROVIDER_VERSION = 0.0.1
 
 default: build
 
-## Check if a 64 bit kernel is running
-UNAME_M := $(shell uname -m)
-
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
 	GOOS += linux
```

Comment:
Hm, why do we do `+= ` here, it doesn't really that well if `$GOOS` is set (same for `GOARCH`).

Also, I wonder why we even do this dance at all. If we only want to ensure that we create a valid `PROVIDER_ARCH` I think we can use `go env` instead:

```shell
$ go env GOOS
darwin
$ go env GOARCH
arm64
```

Reply:
So what I am suggesting is this:

```diff
diff --git a/GNUmakefile b/GNUmakefile
index b6b4fe4..973f0e3 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -4,21 +4,8 @@ PROVIDER_VERSION = 0.0.1

 default: build

-UNAME_S := $(shell uname -s)
-ifeq ($(UNAME_S),Linux)
-       GOOS += linux
-else ifeq ($(UNAME_S),Darwin)
-       GOOS += darwin
-endif
-
-UNAME_M := $(shell uname -m)
-ifeq ($(UNAME_M),x86_64)
-       GOARCH += amd64
-else ifeq ($(UNAME_M),arm64)
-       GOARCH += arm64
-else ifeq ($(UNAME_M),aarch64)
-       GOARCH += arm64
-endif
+GOOS = $(shell go env GOOS)
+GOARCH = $(shell go env GOARCH)
 PROVIDER_ARCH = $(GOOS)_$(GOARCH)

 tools:
@@ -32,9 +19,7 @@ local-clean:
        rm -rf ~/.terraform.d/plugins/localhost/elephantsql/elephantsql/$(PROVIDER_VERSION)/$(PROVIDER_ARCH)/terraform-provider-elephantsql_v$(PROVIDER_VERSION)

 local-build: local-clean
-       @echo $(GOOS);
-       @echo $(GOARCH);
-       GOOS=$(GOOS) GOARCH=$(GOARCH) go build -ldflags "-X 'main.version=$(PROVIDER_VERSION)'" -o terraform-provider-elephantsql_v$(PROVIDER_VERSION)
+       go build -ldflags "-X 'main.version=$(PROVIDER_VERSION)'" -o terraform-provider-elephantsql_v$(PROVIDER_VERSION)

 local-install: local-build
        mkdir -p ~/.terraform.d/plugins/localhost/elephantsql/elephantsql/$(PROVIDER_VERSION)/$(PROVIDER_ARCH)
```

This makes it possible to for other architectures too, like: `GOARCH=amd64 make local-install`.

Reply:
Looks good to me, commit?

Reply:
Pushed in [52e7df4](https://github.com/ElephantSQL/terraform-provider-elephantsql/pull/31/commits/52e7df4cbab9f59b103171c83160868c0faa763a), intentionally did not force push as I wanted to preserve the earlier commit.